data/CourseMemoryDAO.test.js

```diff
@@ -6,14 +7,237 @@ const chai = require('chai');
 
 const should = chai.should();
 const CourseMemoryDAO = require('./CourseMemoryDAO');
+const InMemoryDataStore = require('./InMemoryDataStore');
+
+const Course = require('./Course');
 
 describe('CourseMemoryDAO', () => {
   describe('construction', () => {
-    it('Should initialize with an empty course array and a maxCourseId of 0', () => {
-      /*const dao = new CourseMemoryDAO();
-      chai.expect(dao.courses).to.be.instanceOf(Array);
-      chai.except(dao.courses).to.be.empty();
-      chai.expect(dao.maxCourseId).to.be.equal(0);*/
+    it('Should initialize with access to the in-memory data store object', () => {
+      const dao = new CourseMemoryDAO();
+      chai.expect(dao.data).to.be.equal(InMemoryDataStore);
```

Comment:
in your declarations at the top of this file you can have
`const expect = chai.expect;`

under your chai import and you won't have to do this. Also in this file there is a should being imported that isn't used, but it shouldn't have the parens after it in any case. If you import chai.expect at the top then you can just

`expect(dao.data).to.be.equal(InMemoryDataStore);`

I'm pretty sure with arrays and objects though you have to do `deepEqual()` to do a real comparison.

`expect(thing).to.deepEqual(otherThing);`